core/org.wso2.carbon.tomcat.ext/src/main/java/org/wso2/carbon/tomcat/ext/valves/CompositeValve.java

```diff
@@ -41,8 +41,11 @@ public CompositeValve() {
     @Override
     public void invoke(Request request, Response response) throws IOException, ServletException {
         try {
-            
             if (request.getContext() == null) {
+                if (request.getRequestURI() == null) {
+                    log.debug("Could not handle the request, could be due to not properly encoded request.");
+                    return;
+                }
                 log.error("Could not handle the request, could be due to the maxHttpHeaderSize limitation.");
```

Comment:
this should be also debug log.

Reply:
addressed in 9b2f77e05db33e7cb092432101156e72e13f51d4